File: 26. Remove Duplicates from Sorted Array/C++/Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int> &nums) {
        int i = 0;
        int j = 1;
        int length = nums.size();
        while (j < length) {
            if (nums[i] != nums[j]) {
                if ((i + 1) == j) {
                    i++;
                    j++;
                } else {
                    nums[i + 1] = nums[j];
                    i++;
                    j++;
                }
            } else {
                j++;
            }
        }
        return length == 0 ? i : (i + 1);
    }
};

File: 026. Remove Duplicates from Sorted Array/Python/Remove Duplicates from Sorted Array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = 0
        j = 1
        length = len(nums)
        while j < length:
            if nums[i] != nums[j]:
                if (i + 1) == j:
                    i += 1
                    j += 1
                else:
                    nums[i + 1] = nums[j]
                    i += 1
                    j += 1
            else:
                j += 1
        return i if len(nums) == 0 else i + 1
